x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutoDetectResultProcessor.java

```diff
@@ -462,33 +434,41 @@ private void updateEstablishedModelMemoryOnJob() {
 
         jobResultsProvider.getEstablishedMemoryUsage(jobId, latestBucketTimestamp, modelSizeStatsForCalc, establishedModelMemory -> {
             if (latestEstablishedModelMemory != establishedModelMemory) {
-                updateJob(jobId, Collections.singletonMap(Job.ESTABLISHED_MODEL_MEMORY.getPreferredName(), establishedModelMemory),
-                    new ActionListener<UpdateResponse>() {
-                    @Override
-                    public void onResponse(UpdateResponse response) {
-                        latestEstablishedModelMemory = establishedModelMemory;
-                        LOGGER.debug("[{}] Updated job with established model memory [{}]", jobId, establishedModelMemory);
-                    }
 
-                    @Override
-                    public void onFailure(Exception e) {
-                        LOGGER.error("[" + jobId + "] Failed to update job with new established model memory [" +
-                            establishedModelMemory + "]", e);
+                client.threadPool().executor(MachineLearning.UTILITY_THREAD_POOL_NAME).submit(() -> {
+                    try {
+                        jobUpdateSemaphore.acquire();
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        LOGGER.info("[{}] Interrupted acquiring update established model memory semaphore", jobId);
```

Comment:
level info correct?

Reply:
Hmm, I don't think it's an error level message as the interrupt is not necessarily an error perhaps debug. ml code does not interrupt threads and the only usage I can find in es core outside of tests is in [CancellableThreads](https://github.com/elastic/elasticsearch/blob/6.5/server/src/main/java/org/elasticsearch/common/util/CancellableThreads.java#L165)

I doubt we will every see this message outside of testing. The test framework interrupts zombie threads after each test so that will probably be the only time we see it.